### PR TITLE
Enable performance test (formerly known as a regression test) in CI

### PR DESCRIPTION
**Pull request by chuneuny-emily**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by harimkang** [comment]
@chuneuny-emily Before pushing, be sure to run a pre-commit test with `tox -e pre-commit` and add those changes.

File: tests/regression/conftest.py [hunk 1]
```diff
@@ -46,45 +44,51 @@ def pytest_addoption(parser: pytest.Parser) -> None:
...
+@pytest.fixture(scope="session", autouse=True)
 def fxt_user_name(request: pytest.FixtureRequest) -> str:
     """User name to sign off the regression test execution.
 
     This should be given by the PyTest CLI option.
     """
-    user_name = request.config.getoption("--user-name")
+    user_name = request.config.getoption("--user-name", default=os.environ.get('USER_NAME'))
+    if user_name is None:
+        raise ValueError("Missing user name: Please provide it via the command-line option '--user-name' or set the 'USER_NAME' environment variable.")
```
Comment (vinnamkim) [inline tests/regression/conftest.py]: Why did you change `user_name` and the others as optional arguments from CLI? You should revert this changes and inject the required variables in the GitHub Action run.

File: .github/workflows/perf_test.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Run Performance Test
+        env:
+          MLFLOW_TRACKING_SERVER_URI: ${{ vars.MLFLOW_TRACKING_SERVER_URI }}
+          CI_Perfor_DATASET_ROOT_DIR: ${{ vars.CI_PERFORMANCE_DATASET_ROOT_DIR }}
+          USER_NAME: ${{ vars.USER_NAME }}
+        run: tox -vv -e performance-test
```
Comment (vinnamkim) [inline .github/workflows/perf_test.yaml]:
```suggestion
        run: tox -vv -e performance-test --user-name ${{ vars.USER_NAME }} ...
```

**Comment by chuneuny-emily** [comment]
> @chuneuny-emily Before pushing, be sure to run a pre-commit test with `tox -e pre-commit` and add those changes.

Done

File: tests/regression/conftest.py (hunk 1 above)
Comment (chuneuny-emily) [reply tests/regression/conftest.py]: Done

File: .github/workflows/perf_test.yaml (hunk 2 above)
Comment (chuneuny-emily) [reply .github/workflows/perf_test.yaml]: Done

File: .github/workflows/perf_test.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: V2 Performance Test
+
+on:
+  schedule:
+    - cron: "0 18 * * *" # currently run daily on 18:00 UTC; [TODO]: run weekly
```
Comment (yunchu) [inline .github/workflows/perf_test.yaml]: Generally there would be no changes on the weekends. and currently v1.x daily test are scheduled on UTC 19 so I think the CI resources could be occupied too long  for running daily testing.

```suggestion
    - cron: "0 16 * * 1-5" # currently run daily on 16:00 UTC; [TODO]: run weekly
```

File: .github/workflows/perf_test.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  workflow_dispatch: # run on request
+
+jobs:
+  Performance-Test:
+    runs-on: [self-hosted, linux, x64, dev]
```
Comment (yunchu) [inline .github/workflows/perf_test.yaml]: To use the CI machine that have CI datasets, use "dmount" label instead of "dev"
```suggestion
    runs-on: [self-hosted, linux, x64, dmount]
```

File: .github/workflows/perf_test.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Run Performance Test
+        run: tox -vv -e performance-test -- --mlflow-tracking-uri ${{ vars.MLFLOW_TRACKING_SERVER_URI }} --user-name ${{ vars.USER_NAME }} --dataset-root-dir ${{ vars.CI_PERFORMANCE_DATASET_ROOT_DIR }}
```
Comment (yunchu) [inline .github/workflows/perf_test.yaml]: datasets for v2 would be mounted to the "/home/validation/data/v2" so that could be changed like below
```suggestion
        run: tox -vv -e performance-test -- --mlflow-tracking-uri ${{ vars.MLFLOW_TRACKING_SERVER_URI }} --user-name ${{ vars.USER_NAME }} --dataset-root-dir /home/validation/data/v2
```

**Review by yunchu — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]
LGTM, but it would be good to parallelize multiple tasks in the upcoming PRs. It's highly unlikely for us to run perf tests on multiple Python versions.

File: .github/workflows/perf_test.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Run Performance Test
+        run: tox -vv -e performance-test -- --mlflow-tracking-uri ${{ vars.MLFLOW_TRACKING_SERVER_URI }} --user-name ${{ vars.USER_NAME }} --dataset-root-dir /home/validation/data/v2
```
Comment (vinnamkim) [inline .github/workflows/perf_test.yaml:31]: Why was it changed to the hard-coded path? Can we rollback this to the original one using [Github Action variables](https://docs.github.com/en/actions/learn-github-actions/variables) in the next PR?

```suggestion
        run: tox -vv -e performance-test -- --mlflow-tracking-uri ${{ vars.MLFLOW_TRACKING_SERVER_URI }} --user-name ${{ vars.USER_NAME }} --dataset-root-dir ${{ vars.DATASET_ROOT_DIR }}
```

**Review by vinnamkim — APPROVED** [review]

File: .github/workflows/perf_test.yaml (hunk 6 above)
Comment (chuneuny-emily) [reply .github/workflows/perf_test.yaml:31]: @yunchu suggested it. He said that the dataset folder of the ci machine is automatically mapped to /home/validation/data/ in the ci-runner container, so there is practically no need for it. However, the dataset-root-dir input is required in the CLI, so we keep it. I'll roll-back in a future PR if necessary.
